src/LLMProviders/chatModelManager.ts

```diff
@@ -10,6 +9,35 @@ import { ChatGroq } from "@langchain/groq";
 import { ChatOllama } from "@langchain/ollama";
 import { ChatOpenAI } from "@langchain/openai";
 import { Notice } from "obsidian";
+import { omit, safeFetch } from "@/utils";
+import { ChatAnthropic } from "@langchain/anthropic";
+
+type ChatConstructorType = new (config: any) => BaseChatModel;
+
+// Keep the original ProviderConstructMap type
+type ChatProviderConstructMap = {
```

Comment:
What's this, looks scary 🥶

Reply:
Why not just 

```
const CHAT_PROVIDER_CONSTRUCTORS = {
  [ChatModelProviders.OPENAI]: ChatOpenAI,
  [ChatModelProviders.AZURE_OPENAI]: ChatOpenAI,
  // ... etc
} as const;

type ChatProviderConstructMap = typeof CHAT_PROVIDER_CONSTRUCTORS;
```

Reply:
@logancyang yeah, It's for type Hints.

Reply:
> Why not just
> 
> ```
> const CHAT_PROVIDER_CONSTRUCTORS = {
>   [ChatModelProviders.OPENAI]: ChatOpenAI,
>   [ChatModelProviders.AZURE_OPENAI]: ChatOpenAI,
>   // ... etc
> } as const;
> 
> type ChatProviderConstructMap = typeof CHAT_PROVIDER_CONSTRUCTORS;
> ```

if `ChatModelProviders` add new provider, `CHAT_PROVIDER_CONSTRUCTORS`  don't get any type restriction error. Maybe It's so heavy.